CDCgov/prime-simplereport
# johanna/3850 multiplex results form behavior bugs

# FRONTEND PULL REQUEST

## Related Issue

Fixes #3850 , #3840 , #3851 , #3839 

## Changes Proposed

This PR was branch from Nick Clydes PR. The changes that I introduced are:
- General fixes for styling issues.
- The Test Result form has been renamed to CovidResultForm and a new MultiplexResultForm has been introduced.
- The QueueItem component is a smart component that decides what type of result will be asked in the test card (based on the device selected) and renders the correct result form for it.
- The result forms are dumb components that only care to render the correct UI and perform a lightweight validation to enable and disable the submit button. Once the user makes changes to the result these dumb components notify the smart component Queue Item of the change and wait for it to make the necessary API calls and return back the new persisted state of the result. The dumb components do not hold any result inner state, they are display only.

## Testing

Start a multiplex test and check below:
- When unchecking current date/time to enter custom date the date fields do not move below.
- Check the multiplex info tooltip next to the submit button displays the bubble in a way that the text can be read.
- Check that when checking the results radio buttons and inconclusive check boxes the clicks are properly processed in the UI and that the validation is correct for disabling and enabling the submit button.
 
## Screenshots / Demos

- For large changes, please pair with a designer to ensure changes are as intended

## Checklist for Author and Reviewer

### Design
